presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -1066,6 +1067,61 @@ protected Object visitLikePredicate(LikePredicate node, Object context)
                     optimizedEscape);
         }
 
+        private boolean isLikePattern(Slice pattern, Slice escape)
+        {
+            String stringPattern = pattern.toStringUtf8();
+            if (escape == null) {
+                return stringPattern.contains("%") || stringPattern.contains("_");
+            }
+
+            String stringEscape = escape.toStringUtf8();
+            if (stringEscape.length() != 1) {
+                throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Escape must be empty or a single character");
+            }
+            char escapeChar = stringEscape.charAt(0);
+            boolean escaped = false;
+            for (char currentChar : stringPattern.toCharArray()) {
```

Comment:
So, it actually breaks if the escape character is non-BMP, but fortunately it fails because the Java string is length 2 rather than returning the wrong answer. So we should leave this code as-is because it matches `LikeFunctions.likePattern`.

Add a TODO in both places that this should handle non-BMP Unicode characters and reference the other location, so that someone doesn't change one place but not the other.